src/vw-serving/src/vw_serving/model_manager.py

```diff
@@ -0,0 +1,270 @@
+    def __init__(self):
+        self.sagemaker_tar_gz = self._check_sagemaker_model_dir()
+        self.experiment_id = os.getenv(environment.EXPERIMENT_ID, "default_experiment")
+        self.model_id = os.getenv(environment.MODEL_ID, "default_model")
+
+        self.poll_db = os.getenv(environment.MODEL_METADATA_POLLING, 'true').lower() == 'true'
+
+        if self.poll_db:
+            self._setup_boto_clients()
+
+        self.log_inference_data = os.getenv(
+            environment.LOG_INFERENCE_DATA, 'true').lower() == 'true'
+        if self.log_inference_data:
+            self.firehost_stream = os.getenv(environment.FIREHOSE_STREAM, None)
+            if not self.firehost_stream:
+                raise AlgorithmError(
+                    f"Please specify a firehose stream as '{environment.FIREHOSE_STREAM}' environment variable.")
+
+    def _check_ddb_table_existence(self, ddb_table_resource):
+        # Ensure that the table exists. Throw an error otherwise.
+        try:
+            status = ddb_table_resource.table_status
+            if status != "ACTIVE":
+                raise AlgorithmError(
+                    f"Table: '{ddb_table_resource.name}' is not an 'ACTIVE' table. Its status is {status}")
+        except:
+            raise AlgorithmError(
+                    f"Table: '{ddb_table_resource.name}' does not exist in dynamo db")
+
+    def _setup_boto_clients(self):
+        aws_region = os.getenv("AWS_REGION", "us-west-2")
+        self.boto_session = boto3.Session(region_name=aws_region)
+        
+        # initialize resource clients 
+        self.exp_ddb_table = os.getenv(environment.EXP_METADATA_DYNAMO_TABLE, "")
+        if not self.exp_ddb_table:
+            raise AlgorithmError(
+                    f"Please specify a dynamo db table name as '{environment.EXP_METADATA_DYNAMO_TABLE}' environment variable.")
+        self.model_ddb_table = os.getenv(environment.MODEL_METADATA_DYNAMO_TABLE, "")
+        if not self.model_ddb_table:
+            raise AlgorithmError(
+                    f"Please specify a dynamo db table name as '{environment.MODEL_METADATA_DYNAMO_TABLE}' environment variable.")
+
+        self.exp_ddb_table_resource = self.boto_session.resource('dynamodb').Table(self.exp_ddb_table)
+        self._check_ddb_table_existence(self.exp_ddb_table_resource)
+
+        self.model_ddb_table_resource = self.boto_session.resource('dynamodb').Table(self.model_ddb_table)
+        self._check_ddb_table_existence(self.model_ddb_table_resource)
+
+        self.exp_ddb_wrapper = ExperimentDBClient(self.exp_ddb_table_resource)
+        self.model_ddb_wrapper = ModelDBClient(self.model_ddb_table_resource)
+        self.s3_resource = self.boto_session.resource('s3')
+
+    def _check_sagemaker_model_dir(self):
+        sagemaker_model_path = Path(integ.ARTIFACTS_VOLUME)
+        if sagemaker_model_path.is_dir():
+            model_files = list(sagemaker_model_path.rglob("*"))
+            if len(model_files) >= 2:
```

Comment:
Is it safe to hardcoded as '2' here? Do we control number of generated model files on our end?

Reply:
Yes. For VW there should be atleast 2 files. 'vw.metadata' and 'vw.model' in the artifact directory.